app/imports/ui/pages/EditSmartContract.jsx

```diff
@@ -1,24 +1,47 @@
 import React from 'react';
 import { Grid, Loader, Header, Segment } from 'semantic-ui-react';
 import swal from 'sweetalert';
-import {AutoForm, ErrorsField, LongTextField, NumField, SubmitField, TextField} from 'uniforms-semantic';
+import { AutoForm, ErrorsField, LongTextField, NumField, SubmitField, TextField } from 'uniforms-semantic';
 import { Meteor } from 'meteor/meteor';
 import { withTracker } from 'meteor/react-meteor-data';
 import PropTypes from 'prop-types';
 import SimpleSchema2Bridge from 'uniforms-bridge-simple-schema-2';
+import SimpleSchema from 'simpl-schema';
 import { SmartContracts } from '../../api/smartContract/SmartContract';
 
-const bridge = new SimpleSchema2Bridge(SmartContracts.schema);
+const contractSchema = new SimpleSchema({
+  homeownerName: String,
+  homeownerEmail: String,
+  homeownerPhoneNumber: String,
+  tenantName: String,
+  tenantEmail: String,
+  tenantPhoneNumber: String,
+  unitAddress: String,
+  monthlyRent: Number,
+  termsAndConditions: {
+    type: String,
+    defaultValue: '',
+  },
+});
+
+const bridge = new SimpleSchema2Bridge(contractSchema);
 
 /** Renders the Page for editing a single document. */
 class EditSmartContract extends React.Component {
 
   // On successful submit, insert the data.
   submit(data) {
-    const { homeownerName, homeownerEmail, homeownerPhoneNumber, tenetName, tenetEmail, tenetPhoneNumber, tenetStance, unitAddress, monthlyRent, status, _id } = data;
-    SmartContracts.collection.update(_id, { $set: { homeownerName, homeownerEmail, homeownerPhoneNumber, tenetName, tenetEmail, tenetPhoneNumber, tenetStance, unitAddress, monthlyRent, status } }, (error) => (error ?
-      swal('Error', error.message, 'error') :
-      swal('Success', 'Information updated successfully', 'success')));
+    const { homeownerName, homeownerEmail, homeownerPhoneNumber, tenantName, tenantEmail, tenantPhoneNumber, unitAddress, monthlyRent, status, _id } = data;
+    const username = this.props.user.username;
+
+    if (username === homeownerEmail) {
+      SmartContracts.collection.update(_id, { $set: { homeownerName, homeownerEmail, homeownerPhoneNumber, tenantName, tenantEmail, tenantPhoneNumber, unitAddress,
+        monthlyRent, status } }, (error) => (error ?
+        swal('Error', error.message, 'error') :
+        swal('Success', 'Information updated successfully', 'success')));
+    } else {
+      swal('Error', 'Only homeowner can edit these fields', 'error');
+    }
   }
 
   // If the subscription(s) have been received, render the page, otherwise show a loading icon.
```

Comment:
Optional suggestion: Only display form for homeowners, else display a message "You are not permitted to edit smart contracts."

General comment . If you 're doing single-column grids, you could also just use Containers.